numba/core/ssa.py

```diff
@@ -430,13 +430,17 @@ def _find_def_from_bottom(self, states, label, loc):
             return self._find_def_from_top(states, label, loc=loc)
 
     def _stmt_index(self, defstmt, block, stop=-1):
-        """Find the postitional index of the statement at ``block``.
+        """Find the positional index of the statement at ``block``.
 
         Assumptions:
         - no two statements can point to the same object.
         """
+        # Compare using id() as IR node equality is for semantic equivalence
+        # opposed to direct equality (the location and scope are not considered
+        # as part of the equality measure, this is important here).
+        stmtids = [id(x) for x in block.body]
```

Comment:
The change make sense. However, I am a bit worried about making a fresh list here all the time given how often this is executed and how SSA-reconstruction is already slow. 

I suggesting changing it to a search loop:
```python
        for i in range(len(block.body))[:stop]:
            if block.body[i] is defstmt:
                return i
        return len(block.body)
```

Reply:
Agree. Also makes me wonder about JIT use in the compiler itself :)